Lib/Database/MySqlQuery.php

```diff
@@ -0,0 +1,1247 @@
+      } catch (Exception $e) {
+        throw $e;
+      }
+
+      // Attach conditions to WHERE clause.
+      if ($count != 0)
+        $where .= ', ';
+      $where .= $correct;
+
+      $count++;
+    }
+
+    $where .= ')';
+    $this->where = $where;
+  }
+
+  /**
+   * Build WHERE clause.
+   *
+   * Transform a list of arguments in a string of WHERE conditions.
+   *
+   * @since 1.0.0
+   * @access private
+   *
+   * @param array $clauses `WHERE` clauses.
+   * @param string $operator `AND` or `OR`.
+   *
+   * @throws \RuntimeException if the arguments aren't in the correct form.
+   */
+  private function buildWhereClause(array $clauses, $operator) {
+    $where = $this->where;
+    $method = ($operator == 'AND') ? 'where' : 'orWhere';
+
+    foreach ($clauses as $clause) {
+      if (!is_array($clause))
+        throw new RuntimeException(
+          'Arguments passed to `Query::'.$method.'()` aren\'t in the correct form'
+        );
+
+      if ($where != '')
+        $where = ' '.$operator.' ';
+      else
+        $where = 'WHERE ';
+
+      if (count($clause) == 2) {
+        $value = $clause[1];
+        $oprt = '=';
+      } elseif (count($clause) == 3) {
+        $value = $clause[2];
+        $oprt = $clause[1];
+        if (!self::isOperatorValid($oprt, 'buildWhereClause'))
+          throw new RuntimeException(
+            'Arguments passed to `Query::'.$method.'()` aren\'t in the correct form'
+          );
+      }
+
+      if (is_array($value) && ($oprt == 'between' || $oprt == 'BETWEEN')) {
+        // Replace the value with a string that contains the values in the
+        // array, linked by 'AND'.
+        // If the array doesn't contains exactly 2 values, raise an exception.
+        if (count($value) != 2)
+          throw new RuntimeException(
+            'Arguments passed to `Query::'.$method.'()` aren\'t in the correct form'
+          );
+        try {
+          $value[0] = self::correctValue($value[0]);
+          $value[1] = self::correctValue($value[1]);
+        } catch (Exception $e) {
+          throw $e;
+        }
+        $value = $value[0].' AND '.$value[1];
+      } else {
+        try {
+          $value = self::correctValue($value, $method);
+        } catch (Exception $e) {
+          throw $e;
+        }
+      }
+      // Any other case (e.g. value is integer) is ok as it is.
+
+      $where .= $clause[0].' '.$oprt.' '.$value;
+    }
+
+    $this->where = $where;
+  }
+
+  /**
+   * Correct value.
+   *
+   * @since 1.0.0
+   * @access private
+   * @static
+   *
+   * @param mixed $value Scalar value.
+   * @param string $method Name of the method that called this one (directly or
+   * indirectly).
+   * @return mixed The correct value.
+   *
+   * @throws \RuntimeException if the value is an array.
+   */
+  private static function correctValue($value, $method) {
+    if (is_string($value)) {
+      // Put quotes around the value if it is a string.
+      $value = "'".$value."'";
```

Comment:
I know you don't have a `Driver` instance here, but assuming you have user input in `$value`, you should really escape it to avoid SQL Injection attacks.